python/lolopy/tests/test_learners.py

```diff
@@ -181,6 +181,51 @@ def test_adjust_rtree_learners(self):
         rf.fit(X[:16, :], y[:16])
         self.assertLess(r2_score(y, rf.predict(X)), 1.0)  # Should not fit the whole dataset perfectly
 
+class TestExtraRandomTrees(TestCase):
+
+    def test_extra_random_trees_regressor(self):
+        rf = ExtraRandomTreesRegressor()
+
+        # Train the model
+        X, y = load_boston(True)
+
+        # Make sure we get a NotFittedError
+        with self.assertRaises(NotFittedError):
+            rf.predict(X)
+
+        # Fit the model
+        rf.fit(X, y)
+
+        # Run some predictions
+        y_pred = rf.predict(X)
+        self.assertEqual(len(y_pred), len(y))
+
+        # Test the ability to get importance scores
+        y_import = rf.get_importance_scores(X[:100, :])
+        self.assertEqual((100, len(X)), y_import.shape)
+
+        # Basic test for functionality. R^2 above 0.98 was measured on 27Dec18
+        score = r2_score(y_pred, y)
+        print('R^2:', score)
+        self.assertGreater(score, 0.98)

```

Comment:
This looks like a non-deterministic test.  Plus, I think the threshold was just copied from the RandomForest test; its not clear to me that we'd expect the same from the ExtraRandomTreesRegressor.

Reply:
Good call -- I hadn't updated this since merging in the rng changed. There is still some residual nondeterminism due to threading, so it's not a perfectly deterministic test, but it should be much more reliable now even so.
I have retested the measurements, and they didn't need to change.